auth/auth-server/src/telemetry/quote_comparison/handler.rs

```diff
@@ -72,4 +63,66 @@ impl QuoteComparisonHandler {
         // Execute all futures concurrently and wait for them to complete
         join_all(futures).await;
     }
+
+    /// Records a comparison for a single source
+    async fn record_comparison_for_source(
+        &self,
+        source: QuoteSource,
+        our_quote: &QuoteResponse,
+        side: OrderSide,
+        amount: u128,
+        labels: Vec<(String, String)>,
+    ) -> Result<(), AuthServerError> {
+        let base_token = Token::from_addr(&our_quote.base_mint);
+        let quote_token = Token::from_addr(&our_quote.quote_mint);
+
+        let quote = source.get_quote(base_token, quote_token, side, amount).await;
+        let usdc_per_gas = self.get_usdc_per_gas().await?;
+        let comparison =
+            QuoteComparison { our_quote, source_quote: &quote, gas_usdc_per_gas: usdc_per_gas };
+
+        record_comparison(&comparison, side, &labels);
+        record_output_value_net_of_gas_comparison(&comparison, side, &labels);
+        Ok(())
+    }
+}
+
+impl QuoteComparisonHandler {
+    /// Calculates the USDC cost per unit of gas
+    async fn get_usdc_per_gas(&self) -> Result<f64, AuthServerError> {
+        let gas_price_eth = self.fetch_gas_price_eth().await?;
+
+        let usdc_per_eth = self.fetch_eth_price().await?;
+
+        Ok(usdc_per_eth * gas_price_eth)
+    }
+
+    /// Fetches the current gas price and converts it to ETH
+    async fn fetch_gas_price_eth(&self) -> Result<f64, AuthServerError> {
+        // Fetch gas price in wei
+        let gas_price: U256 = self
+            .arbitrum_client
+            .client()
+            .get_gas_price()
+            .await
+            .map_err(|e| AuthServerError::Custom(e.to_string()))?;
+
+        // Convert wei to eth
+        let eth_string =
+            format_units(gas_price, "eth").map_err(|e| AuthServerError::Custom(e.to_string()))?;
+
+        eth_string.parse::<f64>().map_err(|e| AuthServerError::Custom(e.to_string()))
+    }
+
+    /// Fetches the current price of ETH in USDC
+    async fn fetch_eth_price(&self) -> Result<f64, AuthServerError> {
+        let eth = Token::from_ticker("WETH");
```

Comment:
can we leave a comment here calling out that the price reporter actually fetches the ETH (not WETH) price under the hood